### DISCUSSION TITLE
How to transform html to text?

### DISCUSSION THREAD

**Discussion by RabbitShare** [syntax-tree/unist#125] [Q&A]
How to transform html node to text node?

`<div>` to `\<div>`

**Comment by remcohaszing** [comment]
Can you elaborate what you mean by _html node to text node_? Your example prefixes the `<` with a `\`, which seems odd to me. Are you trying to escape the HTML to insert it raw into the DOM? Something else? Why? What’s the context? Are you using Node.js? A browser?

**Reply by RabbitShare** [reply]
I have a markdown for hackers and it's strongly recommended to escape html tag and show raw html instead of insert it in DOM

**Reply by remcohaszing** [reply]
Ok, I think you’re looking for something like this:

```js
import { visit } from 'unist-util-visit'

/**
 * @type {import('unified').Plugin<[], import('mdast').Root>}
 */
function remarkHtmlToText() {
  return (ast) => {
    visit(ast, 'html', (node, index, parent) => {
      if (index == null || parent == null) {
        return
      }

      parent[index] = {
        type: 'text',
        value: node.value
      }
    })
  }
}
```

**Reply by wooorm** [reply]
@remcohaszing That’s already done by default? Only the `allowDangerousHtml` options will allow HTML?

**Reply by remcohaszing** [reply]
It does? I thought `remark-rehype` strips `html` nodes by default, and `allowDangerousHtml` turns them into `raw` nodes. From what I understand, the OP wants to turn `html` nodes to text.

**Reply by wooorm** [reply]
Ah, yes true for remark-rehype. But thus is in the syntax tree org 🤷‍♂️ maybe micromark or remark-html is used? Nobody knows. OP needs to provide more info
